Automattic/portkey-android
# Video sound toggle MVVM & save

Builds on top of #330 

Closes #325 , not only hiding the sound control view in error mode but also implements the MVVM support for it, so:
- selection for sound on/off is remembered across the frame selector (tapping on different videos should activate/deactivate the sound for each
- passing the sounds toggle state down to the mp4composer so the video gets saved with/without audio as specified
- hiding the control from the error handling view entirely

#### To test:
CASE A: SUCCESSFUL PATH
1. create a story with some videos and images
2. tap sound on/off to toggle audio
3. verify changing the selection "remembers" the audio on/off for each
4. verify selecting an image frame hides the audio toggle
5. tap on publish
6. verify the produced audios have / don't have audio in them

CASE B: UNSUCCESSFUL PATH

0. apply the gist in #300 (or just throw an artificial exception somewhere in method `runPipelines()` in `mp4composerengine` class)
1. perform the same steps as per the successful path
2. once things fail, tap on the error notification to see the story in error handling mode
3. in the error view, verify that no frames show the sound toggle control, but still the different videos will play the audio if it was originally selected to be played, and stay muted if not.
